software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -314,6 +314,7 @@ public void launch() {
 
         if (isPackageInstall()) {
             commands.add(addSbinPathCommand());
+            commands.add(sudo(format("ulimit -n %s", getNumberOfOpenFiles())));
```

Comment:
There will be no problem with setting the hard limit since the ulimit command is executed with elevated privileges (sudo). 

RedHat based distro
```
[root@localhost ~]# cat /etc/redhat-release 
CentOS release 6.6 (Final)
[root@localhost ~]# ulimit -n
1024
[root@localhost ~]# ulimit -n 65536
[root@localhost ~]# ulimit -Sn
65536
[root@localhost ~]# ulimit -Hn
65536
```

Ubuntu
```
# cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.2 LTS"
# ulimit -n 65536
# ulimit -Sn
65536
# ulimit -Hn
65536
```